assets/model_monitoring/components/tests/e2e/test_feature_attribution_drift_signal_monitor_e2e.py

```diff
@@ -175,3 +176,18 @@ def test_monitoring_run_use_defaults_empty_production_data_failed(
 
         # empty target data should cause the pipeline to fail
         assert pipeline_job.status == "Failed"
+
+    def test_monitoring_run_no_target_column_reference_data_successful(
+        self, ml_client: MLClient, get_component, test_suite_name
+    ):
+        """Test the scenario where the production data does not contain the target column."""
+        pipeline_job = _submit_feature_attribution_drift_model_monitor_job(
```

Comment:
I would prefer not to add e2e for this scenario. UT will be better. Because the run time for gated test will be very long if we add a new e2e test for every change.

Reply:
Yes that makes sense. We could maybe also look into parallelizing our e2e tests since most of them are independent from each other.

Reply:
we can bring this to the scrum

Reply:
I think the tests are already parallelized so ignore this comment